src/components/DocumentEditPage/DocumentEditPage.js

```diff
@@ -0,0 +1,105 @@
+import Editor from './Editor.js';
+import DocumentHeader from './DocumentHeader.js';
+import DocumentFooter from './DocumentFooter.js';
+
+import { fetchDocuments } from '../../utils/api.js';
+import {
+  NEW,
+  ROUTE_DOCUMENTS,
+  DEFAULT_DOCUMENT_ID,
+} from '../../utils/constants.js';
+import { isNew, setDocumentTitle } from '../../utils/helper.js';
+import { push } from '../../utils/router.js';
+
+export default function DocumentEditPage({
+  $target,
+  initialState,
+  onDelete,
+  onEdit,
+}) {
+  isNew(new.target);
+
+  const $page = document.createElement('div');
+  $page.className = 'document-edit-page';
+
+  this.state = initialState;
+
+  const documentHeader = new DocumentHeader({
+    $target: $page,
+    initialState: this.state,
+    onDelete,
+  });
+
+  const editor = new Editor({
+    $target: $page,
+    initialState: {
+      title: '',
+      content: '',
+    },
+    onEdit,
+  });
+
+  const documentFooter = new DocumentFooter({
+    $target: $page,
+    initialState: {
+      document: null,
+    },
+  });
+
+  this.setState = async (nextState) => {
+    if (this.state.documentId === nextState.documentId && nextState.document) {
+      this.state = { ...this.state, ...nextState };
+      editor.setState(
+        this.state.document || {
+          title: '',
+          content: '',
+        }
+      );
+      documentHeader.setState(this.state);
+      documentFooter.setState({
+        document: this.state.document,
+      });
+      this.render();
+      return;
+    }
+
+    this.state = { ...this.state, ...nextState };
+
+    if (this.state.documentId === NEW) {
+      editor.setState({
+        title: '',
+        content: '',
+      });
+      documentHeader.setState(this.state);
+      documentFooter.setState({
+        document: null,
+      });
+      this.render();
+    } else {
+      await loadDocument();
+    }
+  };
+
+  const loadDocument = async () => {
+    const document = await fetchDocuments(this.state.documentId);
+    if (!document) {
+      alert('존재하지 않는 페이지입니다. 첫 페이지로 이동합니다.');
+      push(`${ROUTE_DOCUMENTS}/${DEFAULT_DOCUMENT_ID}`);
+      return;
+    }
+
+    this.setState({
+      ...this.state,
+      document,
+    });
+    documentFooter.setState({
+      document: this.state.document,
+    });
+    setDocumentTitle(this.state.document?.title || '');
+  };
+
+  this.render = () => {
+    $target.appendChild($page);
```

Comment:
p3;
이 부분 때문에 에디터에 내용이 수정되면 focusout 되는 것 같아요! 아래와 비슷하게 해당 요소가 존재하지 않는 경우에만 appendChild 해주면 될 것 같습니다!

```suggestion
    if (!$target.querySelector('.document-edit-page')) $target.appendChild($page);
```

Reply:
와 한번에 해결됐어요!! 역시 💯 감사합니다~ eb5b0a4c321325526363918a735acc01ca4e93dd